module/zfs/zfs_dir.c

```diff
@@ -746,9 +749,10 @@ zfs_link_create(zfs_dirlock_t *dl, znode_t *zp, dmu_tx_t *tx, int flag)
 			mutex_exit(&zp->z_lock);
 			return (SET_ERROR(ENOENT));
 		}
-		zp->z_links++;
+		inc_nlink(ZTOI(zp));
```

Comment:
The `I_LINKABLE` warning only triggers if `i_nlink == 0` prior to the increment and is seemingly there to avoid a potential security hole: http://comments.gmane.org/gmane.linux.file-systems/77269

The inode is initialised with `i_nlink = 1` per below, so it's changing to zero between initialisation and here, and I'm trying to work out why that's occurring and if it's reasonable or not.

```
+ zfs_create(dir, dname(dentry), vap, 0, mode, &ip, cr, 0, NULL);
  + zfs_mknode(dzp, vap, tx, cr, 0, &zp, &acl_ids);
  | + zfs_znode_alloc(zsb, db, 0, obj_type, obj, sa_hdl, ZTOI(dzp));
  |   + new_inode(zsb->z_sb);
  |     + new_inode_pseudo(sb);
  |       + alloc_inode(sb);
  |         + inode_init_always(sb, inode)
  |           + inode->__i_nlink = 1;
  + zfs_link_create(dl, zp, tx, ZNEW);
    + inc_nlink(ZTOI(zp));              /* I_LINKABLE warning on i_nlink == 0 */
```